handlers/report/report.go

```diff
@@ -15,9 +15,22 @@ const (
 	LON_PARAM = "lon"
 )
 
-// weatherReportResponse JSON response for GET /weather-report
+// weatherReportResponse JSON response for GET
 type weatherReportResponse struct {
-	Data models.CurrentWeather `json:"data"`
+	Data currentWeather `json:"data"`
```

Comment:
No idea why I decided to embed the model struct here since I've never done that with one of our production services. 

With the pattern I was using, each layer should really have it's own representation of the entity and there should be a shared package of DTOs for the interfaces between the layers. 

I simplified a little by having the business entities (`models`) also serve as the DTOs since we really only have 2 layers we need to pass info between. 

While that's a defensible choice, embedding the models in the responses tightly couples the packages. Also, since the models structs don't have json tags, the attribute keys just get serialized as they're defined on the Go struct and don't match our `snake_case` desired formatting.